utils/submit_tracks_beta2.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import glob
+import os.path
+import requests
+from uuid import UUID
+import yaml
+
+args = argparse.Namespace()
+template_dir = os.path.join(os.path.dirname(__file__), "..", "templates")
+
+def process_input_parameters():
+  parser = argparse.ArgumentParser(description='Submit tracks to Track API based on input datafiles')
+  parser.add_argument('-e', '--env', choices=['local', 'dev', 'staging', 'prod'],
+                      default='dev', help='target Track API deployment (default: %(default)s)')
+  parser.add_argument('-g', '--genome', nargs='*', metavar='GENOME_UUID', help='limit to specific genomes')
+  parser.add_argument('-f', '--file', nargs='*', metavar='FILENAME', help='limit to specific tracks (datafiles)')
+  parser.add_argument('-t', '--template', nargs='*', metavar='TEMPLATE', help='limit to specific track template')
+  parser.add_argument('-d', '--dry-run', action='store_true', help='do not submit tracks, just print them')
+  parser.add_argument('data_dir', metavar='DATA_DIR', help='root directory containing the datafiles (in genome subdirectories)')
+  #repeat tracks: /hps/nobackup/flicek/ensembl/infrastructure/repeats-beta/results
+  #other genomic tracks: /hps/nobackup/flicek/ensembl/infrastructure/arne/e2020-datafile-2024-03
+  parser.parse_args(namespace=args)
+
+# 1) Loop through all the available bigbed/bigwig datafiles
+def process_bigbeds(root_dir: str) -> None:
+  if not os.path.isdir(root_dir):
+    print(f"Error: data directory {root_dir} not found")
+    exit(1)
+  for subdir in os.listdir(root_dir):
+    if not os.path.isdir(f"{root_dir}/{subdir}"):
+      continue
+    try:
+      UUID(subdir, version=4)
+    except ValueError:
+      print(f"Skipping non-genome directory {subdir}")
+      continue
+    if args.genome and subdir not in args.genome:
+      continue
+    print(f"Processing genome {subdir}")
+    for file in os.listdir(f"{root_dir}/{subdir}"):
+      if file.endswith('.bb') or file.endswith('.bw'):
+        find_template(subdir, file)
```

Comment:
Naming. The name `find_template` suggests that it is going to be a function that returns the found template. But it doesn't. It immediately applies the template, and returns nothing.

Reply:
Suggestions? I divided the script to steps (so that production can switch any part, e.g. feed the list of input datafiles from another source than a data dir). This step matches a datafile to its track template and continues with the next step.

Reply:
> Suggestions?

Something like:
```
      if file.endswith('.bb') or file.endswith('.bw'):
        create_track_for_file(subdir, file)
```
?

Reply:
So I should merge `find_template` and `apply_template` (and `submit_track`) to `create_track_for_file` function?

Reply:
But you are already calling `submit_track` from `apply_template`, which in turn you call from within `find_template`. 🤷‍♂️ You could continue doing the same, but rename the top function to something like `create_track_for_file `.

An alternative suggestion is:

```
if file.endswith('.bb') or file.endswith('.bw'):
  template = find_template(subdir, file)
  if template:
    genome_id = subdir
    apply_template(genome_id, file, template)
```

Reply:
Some datafiles will apply/submit multiple templates/tracks, hence the dedicated function for file=>template matching. I renamed it to `match_template` if it's less confusing (although `find_template` is anyway less clear about returning anything than say, `get_template`).